pkg/cloudprovider/aws.go

```diff
@@ -263,7 +279,11 @@ func (a *AWS) getSubnet(networkInterface *ec2.InstanceNetworkInterface) (*net.IP
 		return nil, nil, fmt.Errorf("error: cannot list ec2 subnets, err: %v", err)
 	}
 	if len(describeOutput.Subnets) > 1 {
-		return nil, nil, fmt.Errorf("error: multiple subnets found for the subnet ID: %s", *networkInterface.SubnetId)
+		subnetId := "<nil>"
+		if networkInterface.SubnetId != nil {
```

Comment:
Note: with change in line 196, this is not possible unless getSubnet() starts to be called from another location. Maybe we don't need it here anymore?

Reply:
Have you checked whether `len(describeOutput.Subnets) > 1` can be true when `SubnetId` is set?

Reply:
Looking at the implementation it shouldn't return more than one subnet if there is one non-nil subnet provided but that check is not really hurting us so I am ok either way.